### PR TITLE
Added `decryptMessage` function for implementation `eth_decryptMessage`

### PR DESCRIPTION
**Pull request by logvik**
We need to add it for implementation https://github.com/MetaMask/metamask-extension/issues/1190

### CODE REVIEW

**Comment by logvik** [comment]
What is the reason for such a long delay? 1.5 month passed, can someone speed up this process? I want to do a pull request to Metamask extension, but without this merge, it will be useless. It will let us use new significant features for crypto-chatting.

**Comment by danfinlay** [comment]
Looking to merge this soon, sorry for the long delay. There are still questions about how a user consents to a site's request decryption (is it per packet, or per session, or what?), but we are hoping to fold this into our development branch of our [login permissions](https://ethereum-magicians.org/t/web3-login-permissions/3583) branch [here](https://github.com/MetaMask/metamask-extension/tree/LoginPerSite), as one avenue for user consent that aligns well with a variety of other features we are pursuing to increase dapp access to useful APIs.

**Comment by logvik** [comment]
This PR only for the ability to use decryption message in Metamask extension. In Metamask extension we will have the same window as we have for signing some message. If a user will allow signing, that Dapp will receive a signed message, if not, then the request will be declined. The same for message decryption. We can discuss user interaction flow into Metamask repository.

**Comment by danfinlay** [comment]
Could you fix the conflicts on this PR? Some changes were made to `master`, should be simple things to fix.

**Comment by danfinlay** [comment]
Some good news: Once this is merged, `eth-hd-keyring` will gain the support for the methods automatically, [via inheritance](https://github.com/MetaMask/eth-hd-keyring/pull/12) (just added).

**Comment by logvik** [comment]
@danfinlay the conflicts are resolved. Please check.

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
   "devDependencies": {
     "chai": "^4.1.2",
     "ethereumjs-tx": "^2.1.0",
+    "extend": "^3.0.0",
```
Comment (danfinlay) [inline package.json]: Please remove the unused new dependency.

**Comment by danfinlay** [comment]
Added a CI fix on another PR: https://github.com/MetaMask/eth-simple-keyring/pull/23

**Comment by danfinlay** [comment]
Closing as merged in #23
Included in [version 3.2.0](https://github.com/MetaMask/eth-simple-keyring/pull/24).

Published to npm as version 3.2.0.

**Pull request closed without merging**